source/common/router/config_impl.cc

```diff
@@ -1103,33 +1103,63 @@ RouteMatcher::RouteMatcher(const envoy::config::route::v3::RouteConfiguration& r
   }
 }
 
-RouteConstSharedPtr VirtualHostImpl::getRouteFromEntries(const Http::RequestHeaderMap& headers,
+RouteConstSharedPtr VirtualHostImpl::getRouteFromEntries(const RouteCallback& cb,
+                                                         const Http::RequestHeaderMap& headers,
                                                          const StreamInfo::StreamInfo& stream_info,
                                                          uint64_t random_value) const {
   // No x-forwarded-proto header. This normally only happens when ActiveStream::decodeHeaders
   // bails early (as it rejects a request), so there is no routing is going to happen anyway.
   const auto* forwarded_proto_header = headers.ForwardedProto();
   if (forwarded_proto_header == nullptr) {
+    if (cb) {
+      cb(nullptr, RouteEvalStatus::NoMoreRoutes);
```

Comment:
It's a little odd to me that you don't actually check the return value here and in a few cases below. I think we should check the return value and potentially return nullptr if the callback does not accept it? (In this case there would be no change but int the cases below it does matter.)

Reply:
My bad fixing it